src/topology/bases.lean

```diff
@@ -2,21 +2,25 @@
 Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Mario Carneiro
-
-Bases of topologies. Countability axioms.
 -/
+
 import topology.continuous_on
 
+/-!
+# Bases of topologies. Countability axioms.
+
+## Implementation Notes
+For our applications we are interested that there exists a countable basis, but we do not need the
+concrete basis itself. This allows us to declare these type classes as `Prop` to use them as mixins.
+
```

Comment:
This clearly needs more work, but I'll do it in a separate PR to separate issues. This is the only one of this form I fixed.

Reply:
The other one is `topology/metric_spaces/basic.lean`, which needs a much finer look. Should've split the PR, but didn't expect to find so many small errors in random files first.